.github/workflows/kind-e2e.yaml

```diff
@@ -7,92 +7,195 @@ on:
           # This is attempting to make it a bit clearer what's being tested.
           # See: https://github.com/kubernetes-sigs/kind/releases
-        - cluster: v1.21.1/kourier
-          k8s-version: v1.21.1
-          kind-version: v0.11.1
-          kind-image-sha: sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6
-          kingress: kourier
-          test-flags: "--enable-alpha --enable-beta"
-          cluster-suffix: c${{ github.run_id }}.local
-        - cluster: v1.22.2/istio
-          k8s-version: v1.22.2
-          kind-version: v0.11.1
-          kind-image-sha: sha256:f638a08c1f68fe2a99e724ace6df233a546eaf6713019a0b310130a4f91ebe7f
-          kingress: istio
-          test-flags: "--enable-alpha --enable-beta"
-          cluster-suffix: c${{ github.run_id }}.local
-        - cluster: v1.23.0/contour
-          k8s-version: v1.23.0
-          kind-version: v0.11.1
-          kind-image-sha: sha256:49824ab1727c04e56a21a5d8372a402fcd32ea51ac96a2706a12af38934f81ac
-          kingress: contour
-          test-flags: "--enable-alpha --enable-beta"
-          cluster-suffix: c${{ github.run_id }}.local
-        - cluster: v1.23.0/gateway-api
-          k8s-version: v1.23.0
-          kind-version: v0.11.1
-          kind-image-sha: sha256:49824ab1727c04e56a21a5d8372a402fcd32ea51ac96a2706a12af38934f81ac
-          kingress: gateway-api
-          test-flags: "--enable-alpha"
-          cluster-suffix: c${{ github.run_id }}.local
+          #      https://hub.docker.com/r/kindest/node/tags
+        - k8s-version: v1.22.7
+          kind-version: v0.12.0
+          kind-image-sha: sha256:1dfd72d193bf7da64765fd2f2898f78663b9ba366c2aa74be1fd7498a1873166
+
+        - k8s-version: v1.23.5
+          kind-version: v0.12.0
+          kind-image-sha: sha256:a69c29d3d502635369a5fe92d8e503c09581fcd406ba6598acc5d80ff5ba81b1
+
+        # Disabled due to consistent failures
+        # - ingress: gateway_istio
+        #   ingress-class: gateway-api
+        #   test-flags: -enable-alpha
+        #   namespace-resources: httproute
+
+        - ingress: contour
+          namespace-resources: httpproxy
+
+        - ingress: istio
+          namespace-resources: virtualservices
+
+        - test-suite: runtime
+          test-path: ./test/conformance/runtime/...
+
+        - test-suite: api
+          test-path: ./test/conformance/api/...
+
+        - test-suite: e2e
+          test-path: ./test/e2e
 
     env:
-      GOPATH: ${{ github.workspace }}
-      GO111MODULE: on
-      GOFLAGS: -tags=nostackdriver
-      # https://github.com/google/go-containerregistry/pull/125 allows insecure registry for
-      # '*.local' hostnames. This works both for `ko` and our own tag-to-digest resolution logic,
-      # thus allowing us to test without bypassing tag-to-digest resolution.
-      REGISTRY_NAME: registry.local
-      REGISTRY_PORT: 5000
-      KO_DOCKER_REPO: registry.local:5000/knative
+      SYSTEM_NAMESPACE: knative-serving
```

Comment:
Otherwise this is a random GUID (because we source `test/e2e-common.sh`)

Reply:
but don't we need the random generated namespace?

Reply:
No - what for?

The intent from what I recall is to make sure our components work in a non-default namespace. Which they do and our prow e2e still exercise that behaviour.

I'm good with going either way here

Reply:
Is there any chance that the Prow e2e test won't get ported when we migrate over to Github Actions? If so, then it might make sense to go ahead and use the random namespace here.

Reply:
+1 to Paul's comment here

Reply:
I'm good to remove this - let me do it